### PR TITLE
Replace error log levels with warnings in tedge agent

### PR DESCRIPTION
**Pull request by Ruadhri17**
## Proposed changes
This PR replaces error log levels with warnings and prevents exiting with error code in such cases.

## Types of changes

- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Improvement (general improvements like code refactoring that doesn't explicitly fix a bug or add any new functionality)
- [ ] Documentation Update (if none of the other choices apply)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Paste Link to the issue
* #2634 

## Checklist

- [x] I have read the [CONTRIBUTING](https://github.com/thin-edge/thin-edge.io/blob/main/CONTRIBUTING.md) doc
- [x] I have signed the [CLA](https://github.com/thin-edge/thin-edge.io/blob/main/CONTRIBUTOR-LICENSE-AGREEMENT.md) (in all commits with git commit -s)
- [x] I ran `cargo fmt` as mentioned in [CODING_GUIDELINES](https://github.com/thin-edge/thin-edge.io/blob/main/CODING_GUIDELINES.md)
- [x] I used `cargo clippy` as mentioned in [CODING_GUIDELINES](https://github.com/thin-edge/thin-edge.io/blob/main/CODING_GUIDELINES.md)
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)

### CODE REVIEW

File: crates/core/tedge_agent/src/software_manager/actor.rs [hunk 1]
```diff
@@ -107,10 +107,10 @@ impl Actor for SoftwareManagerActor {
             tokio::select! {
                 _ = self.handle_request(request, &mut plugins, &operation_logs) => {
                     if let Err(SoftwareManagerError::NotRunningLatestVersion) = Self::detect_self_update() {
-                        error!("Tedge-agent is no more running the latest-version => a restart is required");
+                        warn!("Tedge-agent is no more running the latest-version => a restart is required");
                         // Make sure the operation status is properly reported before the restart
                         tokio::time::sleep(Duration::from_secs(5)).await;
-                        return Err(RuntimeError::ActorError(Box::new(SoftwareManagerError::NotRunningLatestVersion)));
+                        break;
```
Comment (didier-wenzek) [inline crates/core/tedge_agent/src/software_manager/actor.rs:113]: Oops, returning an error is critical here. This line must be reverted. Only the log levels has to be changed.

**Review by didier-wenzek — CHANGES_REQUESTED** [review]
@albinsuresh I wonder why the test ["Update tedge version from base to current using Cumulocity"](https://github.com/thin-edge/thin-edge.io/blob/main/tests/RobotFramework/tests/cumulocity/self-update/tedge_self_update.robot#L95) is not failing as it should since the version mismatch is not reported after an update.

**Comment by albinsuresh** [comment]
> @albinsuresh I wonder why the test ["Update tedge version from base to current using Cumulocity"](https://github.com/thin-edge/thin-edge.io/blob/main/tests/RobotFramework/tests/cumulocity/self-update/tedge_self_update.robot#L95) is not failing as it should since the version mismatch is not reported after an update.

@didier-wenzek Is this comment still valid as Krzystoff seem to have reverted the problematic change?

**Comment by didier-wenzek** [comment]
> > @albinsuresh I wonder why the test ["Update tedge version from base to current using Cumulocity"](https://github.com/thin-edge/thin-edge.io/blob/main/tests/RobotFramework/tests/cumulocity/self-update/tedge_self_update.robot#L95) is not failing as it should since the version mismatch is not reported after an update.
> 
> @didier-wenzek Is this comment still valid as Krzystoff seem to have reverted the problematic change?

Somehow, yes. This test was supposed to cover the feature broken by the problematic change, but failed to raise the issue.

That said this is unrelated to this PR.

**Review by didier-wenzek — APPROVED** [review]
Approved

**Comment by albinsuresh** [comment]
> > > @albinsuresh I wonder why the test ["Update tedge version from base to current using Cumulocity"](https://github.com/thin-edge/thin-edge.io/blob/main/tests/RobotFramework/tests/cumulocity/self-update/tedge_self_update.robot#L95) is not failing as it should since the version mismatch is not reported after an update.
> > 
> > 
> > @didier-wenzek Is this comment still valid as Krzystoff seem to have reverted the problematic change?
> 
> Somehow, yes. This test was supposed to cover the feature broken by the problematic change, but failed to raise the issue.
> 
> That said this is unrelated to this PR.

@didier-wenzek That's because his changes were not included in the `0.0.1` dummy version from which we upgrade to the current version in the test.

I understand how this can be confusing to anyone. Upgrading from the current version to a dummy higher version would have been better and clearer. We resorted to this mechanism due to the complexity of always building that dummy higher version as part of the test setup. We can re-evaluate that choice post 1.0.

File: crates/core/tedge_agent/src/state_repository/state.rs [hunk 2]
```diff
@@ -36,7 +36,7 @@ impl<T: DeserializeOwned + Serialize> AgentStateRepository<T> {
                     state_dir
                 }
                 Err(err) => {
-                    error!("Cannot use {state_dir:?} to store tedge-agent state: {err}");
+                    warn!("Cannot use {state_dir:?} to store tedge-agent state: {err}");
```
[the remark this answers is not in the document]
Comment (didier-wenzek) [reply crates/core/tedge_agent/src/state_repository/state.rs:39]: A warning sounds appropriate. Yes those not using the feature will have this on every start-up but this is a key point for those who want to use the feature. This is even for that reason that I used an error level at first.

[the remark this answers is not in the document]
Comment (reubenmiller) [reply crates/core/tedge_agent/src/state_repository/state.rs:39]: ok, then leave it as warning.

**Review by reubenmiller — APPROVED** [review]

**Pull request merged**